continuous_integration/check_no_test_skipped.py

```diff
@@ -11,26 +11,34 @@
 
 # dict {test: result} where result is False if the test was skipped in every
 # job and True otherwise.
-aggregated_results = {}
+always_skipped = {}
 
 for name in os.listdir(base_dir):
     # all test result files are in /base_dir/jobs.*/ dirs
     if name.startswith("stage1."):
+        print("> processing test result from job", name.replace("stage1", ""))
+        print("  > tests skipped:")
         result_file = os.path.join(base_dir, name, "test-data.xml")
         root = ET.parse(result_file).getroot()
 
         # All tests are identified by the xml tag testcase.
-        for test in root.iter('testcase'):
-            test_name = test.attrib['name']
-            if test_name not in aggregated_results:
-                # len(test) is > 0 if the test is skipped.
-                aggregated_results[test_name] = not bool(len(test))
+        for test in root.iter("testcase"):
+            test_name = test.attrib["name"]
+            if test_name not in always_skipped:
+                always_skipped[test_name] = False
+
+            for child in test:
+                if child.tag == "skipped":
+                    print("    -", test_name)
+                    always_skipped[test_name] |= True
+                    break
             else:
-                aggregated_results[test_name] |= not bool(len(test))
+                always_skipped[test_name] = False
 
+print("\n------------------------------------------------------------------\n")
 fail = False
-for test, result in aggregated_results.items():
-    if not result:
+for test, skipped in always_skipped.items():
```

Comment:
Not sure if necessary, but never hurts:

```suggestion
for test, skipped in sorted(always_skipped.items()):
```

Reply:
hum why do you want to reorder the tests ? I find it easier is they keep the order they are run in the test suite.

Reply:
Alright.